src/Swift.Bindings/src/Emitter/StringCSharpEmitter/Handler/MethodHandler.cs

```diff
@@ -325,7 +347,14 @@ public void HandleReturnType()
         {
             if (!MarshallingHelpers.MethodRequiresIndirectResult(MethodDecl, ParentDecl, TypeDatabase))
             {
-                var returnType = MethodDecl.CSSignature.First().CSTypeIdentifier.Name;
+                var argument = MethodDecl.CSSignature.First();
+                var returnType = argument.CSTypeIdentifier.Name;
+                var typeRecord = MarshallingHelpers.GetType(argument, TypeDatabase.Registrar);
+                if (typeRecord == null || !typeRecord.IsProcessed)
```

Comment:
All types should be registered during ABI parsing, at least for lazy-loading. If a type is processed, it means the tooling will generate a projection type; otherwise, it will emit an "AnyType" placeholder. If a type can't be found in the database, it indicates that it wasn't encountered during ABI parsing in any form, and throwing an exception is fine as it is bug.

Reply:
But then the 'typeRecord == null' part becomes unnecessary (error will be thrown before reaching it). I think we should not make GetType return nullable if null result means critical failure.

Reply:
I agree.

Reply:
Reverted. This is how I discovered a type registration issue :)